src/cscapi/storage.py

```diff
@@ -97,7 +97,7 @@ def get_signals(
         raise NotImplementedError
 
     @abstractmethod
-    def get_machine_by_id(self, machine_id: str) -> MachineModel:
+    def get_machine_by_id(self, machine_id: str) -> Optional[MachineModel]:
```

Comment:
sql and mongodb storage implementation already returns Optional[MachineModel] ; 
None is returned when the machine does not exist in database